addons/product/models/product_product.py

```diff
@@ -425,6 +435,221 @@ def _onchange_uom_id(self):
+        if not import_attribute_values:
+            return super()._load_records_write(values)
+
+        attribute_values = import_attribute_values.split(',')
+        existing_values = self.import_attribute_values.split(',')
+        wrong = len(attribute_values) != len(existing_values)
+        if not wrong:
+            for value in attribute_values:
+                attribute_name, value_name = value.split(':', 1)
+                wrong = f'{attribute_name.strip()}:{value_name.strip()}' not in existing_values
+        if wrong:
+            raise ValueError('The exitings product has different attribute value.'
+                f'{import_attribute_values!r} is not equivalent to {",".join(existing_values)!r} for {self.get_external_id()[self.id]!r}, {self.id}')
+
+        if import_attribute_values:
+            for key in list(values):
+                if not values[key]:
+                    values.pop(key)
+
+        return super()._load_records_write(values)
+
+    def _load_records_create(self, data_list):
+        with_import_values = [vals for vals in data_list if vals.get('import_attribute_values')]
+        if not with_import_values:
+            return super()._load_records_create(data_list)
+
+        for vals in with_import_values:
+            vals['name'] = (vals.get('name') or '').strip()
+
+        if any(not vals['name'] and not vals.get('product_tmpl_id') for vals in with_import_values):
+            raise ValueError('Unable to import products with attribute values but without name of product set')
+
+        PT = self.env['product.template'].with_context(create_product_product=False, update_product_template_attribute_values=True).sudo()
+        PA = PT.env['product.attribute']
+        PAV = PT.env['product.attribute.value']
+        PTAL = PT.env['product.template.attribute.line']
+
+        ###############################################
+        # Search product.attribute.value or create it #
+        ###############################################
+
+        attribute_to_values = defaultdict(set)  # attribute_name => list[value_name]
+        attribute_to_values[''] = set()
+        for vals in with_import_values:
+            for value in vals['import_attribute_values'].split(','):
+                if ':' not in value:
+                    raise ValueError(f'Unable to import products with attribute value without attribute name (defined as: attribute:value): {value}')
+                attribute_name, value_name = value.split(':', 1)
+                attribute_to_values[attribute_name.strip()].add(value_name.strip())
+
+        pa_pav_records = {}  # (attribute_name, value_name) => Record<'product.attribute.value'>
+        pa_records = {}  # attribute_name => Record<'product.attribute'>
+        values_to_create = []
+        for attribute_name in reversed(attribute_to_values):  # search for and create the value without attributes last
+            value_names = attribute_to_values[attribute_name]
```

Comment:
This is a `set`.  Beware of nondeterminism!
```suggestion
            value_names = sorted(attribute_to_values[attribute_name])
```